test/Flick/AnimatedCarousalPage.xaml.cs

```diff
@@ -49,6 +57,76 @@ public sealed partial class AnimatedCarousalPage : Page
         public AnimatedCarousalPage()
         {
             this.InitializeComponent();
+
+            ElementCompositionPreview.GetElementVisual(sv).Clip = ElementCompositionPreview.GetScrollViewerManipulationPropertySet(sv).Compositor.CreateInsetClip();
+        }
+
+        private object selectedItem = null;
+
+        public object SelectedItem
+        {
+            get
+            {
+                return selectedItem;
+            }
+            set
+            {
+                selectedItem = value;
+            }
+        }
+
+        private bool isScrolling = false;
+
+        public bool IsScrolling
+        {
+            get
+            {
+                return isScrolling;
+            }
+        }
+
+        private double itemScaleRatio = 0.5;
+
+        public double ItemScaleRatio
+        {
+            get
+            {
+                return itemScaleRatio;
+            }
+            set
+            {
+                itemScaleRatio = value;
+            }
+        }
+
+        protected void OnScrollViewerViewChanged(object sender, ScrollViewerViewChangedEventArgs e)
+        {
+            var svCenterPoint = sv.HorizontalOffset + sv.ViewportWidth / 2;
+            int selectedItemIndex = (int)Math.Floor((svCenterPoint + layout.Spacing / 2) / (layout.Spacing + layout.ItemWidth));
+            selectedItemIndex %= ((System.Collections.Generic.IReadOnlyList<object>)repeater.ItemsSource).Count;
+            var selectedUIElement = repeater.TryGetElement(selectedItemIndex);
+            SelectedItem = (selectedUIElement == null ? null : ((UserControl)selectedUIElement).DataContext);
```

Comment:
you probably don't want to constantly change the SelectedItem while scrolling. This block can move into the else clause below.

Reply:
I'm not changing the selecteditem while scrolling. It only gets updated when scrolling **stops**. Which is the desired functionality of a carousel control with selection support.

Reply:
according to this code, SelectedItem is changing during scrolling because it is outside of the check for e.IsIntermediate.

---
In reply to: [287160617](https://github.com/microsoft/microsoft-ui-xaml/pull/743#discussion_r287160617) [](ancestors = 287160617)